components/zcash_note_encryption/src/lib.rs

```diff
@@ -514,8 +527,9 @@ pub fn try_note_decryption<D: Domain, Output: ShieldedOutput<D, ENC_CIPHERTEXT_S
 ) -> Option<(D::Note, D::Recipient, D::Memo)> {
```

Comment:
Does it make sense for the Domain trait always take a prepared IVK everywhere?

Reply:
@daira and I finished looking into this today, and established that we needed a way to use a prepared scalar for both `CofactorGroup` and `PrimeGroup` multiplication. The `Wnaf` struct is insufficient for this, so I've opened another PR on `group` (https://github.com/zkcrypto/group/pull/36) that alters the previously-introduced APIs to enable this.

Reply:
In the end, it turned out that we could just use `Domain::IncomingViewingKey` _as_ the prepared IVK everywhere.